src/intel_pmu.c

```diff
@@ -204,6 +204,11 @@ static int pmu_config_hw_events(oconfig_item_t *ci) {
     return -EINVAL;
   }
 
+  if (g_ctx.hw_events) {
```

Comment:
Can this even happen? configuration only ever gets called once?

Reply:
Maybe that will be a good idea to allow multiple occurrences of this option, so each occurence will enable own list of events?

```
<Plugin intel_pmu>
           ...
           HardwareEvents "L2_RQSTS.CODE_RD_HIT,L2_RQSTS.CODE_RD_MISS" 
           HardwareEvents "L2_RQSTS.ALL_CODE_RD"
...
         </Plugin>
```
That will be easier to comment/uncomment options then. )

That do you think?

Reply:
Not bad idea, but it can brake the backward compatibility. The other way is to keep both possibilities together, but it increases complexity. What do you think @maryamtahhan ?
Btw. it is better to add it as a separate improvement, as it is not part of the bug fix.

Reply:
>Not bad idea, but it can brake the backward compatibility.

Why it will break backward compatibility? I think it could be implemented without it.

Reply:
So, people can enable a lot of these Hardware Events (https://perf.wiki.kernel.org/index.php/Tutorial#Events).... I have no objection to the suggestion by @rpv-tomsk  as long as the comma separated list is maintained as @kwiatrox suggests as this plugin is already widely used.

Also Ok with this being a separate change.